mm/2s2h/Enhancements/Minigames/Archery.cpp

```diff
@@ -51,4 +59,31 @@ void RegisterArchery() {
     });
 }
 
-static RegisterShipInitFunc initFunc(RegisterArchery, { SWAMP_CVAR_NAME, TOWN_CVAR_NAME });
+static void ResetBoatArcheryScore(UNUSED Actor* actor) {
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_25_20)) {
+        HS_SET_BOAT_ARCHERY_HIGH_SCORE(BOAT_CVAR - 1);
+    }
+}
+
+static void RegisterBoatArchery() {
+    COND_ID_HOOK(OnActorInit, ACTOR_EN_DNH, BOAT_CVAR != 20, ResetBoatArcheryScore);
+}
+
+static bool ShouldFailBoatArchery() {
+    return gSaveContext.minigameHiddenScore >= BOAT_HEALTH_CVAR;
+}
+
+static void RegisterKoumeHealth() {
+    REGISTER_VB_SHOULD(VB_FAIL_BOAT_ARCHERY, { *should = ShouldFailBoatArchery(); });
+}
```

Comment:
I'd use `COND_VB_SHOULD`, conditional on `BOAT_HEALTH_CVAR != 10` since that's the vanilla default. 
```suggestion
static void RegisterKoumeHealth() {
    COND_VB_SHOULD(VB_FAIL_BOAT_ARCHERY, BOAT_HEALTH_CVAR != 10, { *should = ShouldFailBoatArchery(); });
}
```

Reply:
Because the comparison is made inside the hook, we don't have to re-register this hook every time the CVar changes, which could be a lot of times because it's controlled by a slider.

Reply:
You make a valid point, but the trade off is this still runs every frame during the minigame even when its set to the vanilla default. Your concern during slider drag I don't disagree with, but the cost is negligible since its only during the use of the slider. The overhead at default is my main concern, but I'm not going to say you have to change it. It just goes against the established pattern in most cases I've seen in the codebase.